src/PrestaShopBundle/Form/Admin/Sell/Product/Shipping/ShippingType.php

```diff
@@ -57,24 +58,32 @@ class ShippingType extends TranslatorAwareType
      */
     private $deliveryTimeNoteTypesProvider;
 
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
     /**
      * @param TranslatorInterface $translator
      * @param array $locales
      * @param string $currencyIsoCode
      * @param FormChoiceProviderInterface $carrierChoiceProvider
      * @param FormChoiceProviderInterface $additionalDeliveryTimeNoteTypesProvider
+     * @param RouterInterface $router
      */
     public function __construct(
         TranslatorInterface $translator,
         array $locales,
         string $currencyIsoCode,
         FormChoiceProviderInterface $carrierChoiceProvider,
-        FormChoiceProviderInterface $additionalDeliveryTimeNoteTypesProvider
+        FormChoiceProviderInterface $additionalDeliveryTimeNoteTypesProvider,
+        RouterInterface $router
```

Comment:
Could you define the BC Break ? 

The parameter doesn't exist in 178x : https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/src/PrestaShopBundle/Form/Admin/Sell/Product/Shipping/ShippingType.php